src/passes/OptimizeInstructions.cpp

```diff
@@ -533,9 +534,16 @@ struct OptimizeInstructions : public WalkerPass<PostWalker<OptimizeInstructions,
               } else if (left->op == OrInt32) {
                 leftRight->value = leftRight->value.or_(right->value);
                 return left;
-              } else if (left->op == ShlInt32 || left->op == ShrUInt32 || left->op == ShrSInt32) {
-                leftRight->value = leftRight->value.add(right->value);
-                return left;
+              } else if (left->op == ShlInt32 || left->op == ShrUInt32 || left->op == ShrSInt32 ||
```

Comment:
I think ideally it'd be a python-y `elif left.op in [ShlInt32, ShrUInt32, ...]:`, though that's less nice to express in C++.
I agree that a switch is syntactically more gross, and stacking `case x: case y: case z:` is unpleasant. [Goodness knows the evils that people can do unto each other via switch statements](https://github.com/bpowers/musl/blob/master/src/stdio/vfprintf.c#L570). The aesthetic preference I'm aiming for here is reducing the repeated `left->op ==`s, to get a better "semantic meaning per syntax token" ratio.

But the key word there is probably "preference", so I think it's fine as-is.